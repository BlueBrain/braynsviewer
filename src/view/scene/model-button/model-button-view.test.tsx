// To test a React component, you need to install these modules:
// yarn add --dev react-test-renderer @types/react-test-renderer
// @see https://jestjs.io/docs/en/snapshot-testing
//
// If a test failed just because you intended to improve the component,
// just call `jest --updateSnapshot`.

import React from 'react'
import Renderer from 'react-test-renderer'
import ModelButtonView, { ModelButtonViewProps } from './model-button-view'

function view(partialProps: Partial<ModelButtonViewProps>) {
    const props: ModelButtonViewProps = {
        model: {
            id: 666,
            name: "My model",
            path: "/gpfs/somewhere/in/this/huge/data/storage/BlueConfig"
        },
        ...partialProps
    }
    return Renderer.create(<ModelButtonView {...props} />).toJSON()
}

describe('<ModelButtonView/> in view/scene', () => {
    it('should be consistent with previous snapshot', () => {
        expect(view({})).toMatchSnapshot()
    })
})
