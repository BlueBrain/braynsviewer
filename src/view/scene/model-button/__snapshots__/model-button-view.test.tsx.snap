// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModelButtonView/> in view/scene should be consistent with previous snapshot 1`] = `
<button
  className="custom view-scene-ModelButtonView"
>
  <div>
    <div>
      My model
    </div>
    <div>
      666
    </div>
  </div>
  <div
    title="/gpfs/somewhere/in/this/huge/data/storage/BlueConfig"
  >
    /gpfs/somewhere/in/this/huge/data/storage/BlueConfig
  </div>
</button>
`;
